# input
string state
string kind
string level
---
# output
bool success
int8[] do_step
int8[] remove_step
string win_side
